File: UIKit and Animations/WWDC 2014/Building Adaptive Apps with UIKit/AdaptivePhotos An Adaptive Application/AdaptivePhotosAnAdaptiveApplication/AdaptivePhotos/AAPLPhoto.m
/*
 Copyright (C) 2014 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 
  The model object that represents an individual photo.
  
 */

#import "AAPLPhoto.h"

@implementation AAPLPhoto

+ (instancetype)photoWithDictionary:(NSDictionary *)dictionary
{
    AAPLPhoto *photo = [[self alloc] init];
    photo.imageName = [dictionary objectForKey:@"imageName"];
    photo.comment = [dictionary objectForKey:@"comment"];
    photo.rating = [[dictionary objectForKey:@"rating"] integerValue];
    return photo;
}

- (UIImage *)image
{
    return [UIImage imageNamed:self.imageName];
}

@end

File: UIKit and Animations/WWDC 2014/Building Adaptive Apps with UIKit/AdaptivePhotos An Adaptive Application/AdaptivePhotos_Swift/AdaptivePhotos_Swift/Model/AAPLPhoto.swift
/*
Copyright (C) 2014 Apple Inc. All Rights Reserved.
See LICENSE.txt for this sample’s licensing information

Abstract:

The model object that represents an individual photo.

*/

import UIKit

class AAPLPhoto: NSObject {
	
	var image: UIImage! {
		let image = UIImage(named: self.imageName)
		return image
	}
	
	var imageName: String!
	var comment: String!
	var rating: Int!
	
	static func photoWithDictionary(_ dictionary: [String : Any]) -> AAPLPhoto {
		let photo = AAPLPhoto()
		photo.imageName = dictionary["imageName"] as? String
		photo.comment = dictionary["comment"] as? String
		photo.rating = dictionary["rating"] as? Int
		return photo
	}
	
}
